src/core/InstrumentFunctions.cpp

```diff
@@ -395,15 +399,48 @@ void InstrumentFunctionArpeggio::processNote( NotePlayHandle * _n )
 
 		frames_processed += remaining_frames_for_cur_arp;
 
-		// in sorted mode: is it our turn or do we have to be quiet for
-		// now?
-		if( static_cast<ArpMode>(m_arpModeModel.value()) == ArpMode::Sort &&
-				( ( cur_frame / arp_frames ) % total_range ) / range != (f_cnt_t) _n->index() )
+		// for sort mode, we combine all of the chord's keys for every
+		// currently playing note in an arrray, sort it and play it
+		// sortOffset offsets range to account for
+		// the arp getting bigger if arpmode = sort
+		int sortOffset = 0;
+		// this array will contain the combined, sorted keys
+		std::vector<int> noteKeysArray(total_chord_size);
+		if (static_cast<ArpMode>(m_arpModeModel.value()) == ArpMode::Sort)
 		{
-			// update counters
-			frames_processed += arp_frames;
-			cur_frame += arp_frames;
-			continue;
+			int minIndex = _n->index();
+
+			// combining all of the chord keys
+			for (int i = 0; i < cnphv.size(); i++)
+			{
+				for (int j = 0; j < cur_chord_size; j++)
+				{
+					// get the value of the chord's selected note (j)
+					// + the base note's value
+					noteKeysArray[i * cur_chord_size + j] = cnphv[i]->key() + chord_table.chords()[selected_arp][j];
+				}
+				if (cnphv[i]->index() < minIndex)
+				{
+					minIndex = cnphv[i]->index();
+				}
+			}
+
+			// avoid playing same key for all
+			// currently playing notes
+			if (minIndex != _n->index())
+			{
+				break;
+			}
+
+			// sorting:
+			std::sort(noteKeysArray.begin(), noteKeysArray.end(), [](int a, int b)
+			{
+				return a < b;
+			});
```

Comment:
Is it possible to just use `std::less`?
```suggestion
			std::sort(noteKeysArray.begin(), noteKeysArray.end(), std::less{});
```

Reply:
Replaced lambda with std::less.